python/Ganga/Core/InternalServices/ShutdownManager.py

```diff
@@ -114,6 +114,10 @@ def add_priority(x):
 
     logger.info("Stopping Job processing before shutting down Repositories")
 
+    from Ganga.Core.GangaThread.WorkerThreads import _global_queues
+    if _global_queues:
+        _global_queues.lock()
```

Comment:
Frankly the function need renaming but I'll get around to this, probably in this PR, I suspect ```freeze``` more correctly matches what is going on here anyway.